### PR TITLE
[CTOR-46] [Plugin] Add Pack Teldat SNMP 

### PR DESCRIPTION
**Pull request by lucie-dubrunfaut**
## Description

Add Teldat SNMP plugin with following modes : 
- CPU
- Mémoire
- Traffic
- Hardware
- Cells-Radio

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (develop).

### CODE REVIEW

**Review by sc979 — DISMISSED** [review]

File: src/network/teldat/snmp/mode/interfaces.pm [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+
+__END__
+
+=head1 MODE
+
+Check interfaces.
+
+=over 8
+
+=item B<--add-global>
```
Comment (Evan-Adam) [inline src/network/teldat/snmp/mode/interfaces.pm:46]: Aren't these options already documented in the interfaces.pm file ?

**Review by Evan-Adam — CHANGES_REQUESTED** [review]
pkg.json seem to have a typo

File: src/network/teldat/snmp/mode/interfaces.pm (hunk 1 above)
Comment (lucie-dubrunfaut) [reply src/network/teldat/snmp/mode/interfaces.pm:46]: To know this you can try interfaces command with help option and compare results with and without this section.

**Review by sdepassio — APPROVED** [review]

File: src/network/teldat/snmp/plugin.pm [hunk 2]
```diff
@@ -0,0 +1,52 @@
+#
+# Copyright 2023 Centreon (http://www.centreon.com/)
```
[the remark this answers is not in the document]
Comment (omercier) [reply src/network/teldat/snmp/plugin.pm:2]: Good remark! I've created a ticket to manage that :+1:

File: src/network/teldat/snmp/mode/cellsradio.pm [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+=item B<--critical-status>
+
+Define the conditions to match for the status to be CRITICAL (default: '%{simStatus} =~ /LOCKED/ || %{simStatus} =~ /DETECTING/').
+You can use the following variables: %{simStatus}, %{interfaceState}, %{cellId}, %{simIcc}, %{operator}, %{imsi}
+
+=item B<--warning-*> B<--critical-*>
+
+Thresholds.
+Can be: 'modules-cellradio-detected', 'module-cellradio-rsrp', ''module-cellradio-rsrq', 'module-cellradio-rscp', 'module-cellradio-csq'
```
[the remark this answers is not in the document]
Comment (lucie-dubrunfaut) [reply src/network/teldat/snmp/mode/cellsradio.pm:341]: ok

File: src/network/teldat/snmp/mode/cpu.pm [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        }
+    ];
+}
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options, force_new_perfdata => 1);
+    bless $self, $class;
+
+    $options{options}->add_options(arguments => {
```
[the remark this answers is not in the document]
Comment (lucie-dubrunfaut) [reply src/network/teldat/snmp/mode/cpu.pm:74]: Initialize options => see this [code search](https://github.com/search?q=repo%3Acentreon%2Fcentreon-plugins%20%20%24options%7Boptions%7D-%3Eadd_options(arguments%20%3D%3E%20%7B&type=code)

**Review by omercier — APPROVED** [review]
Tests passed

**Comment by omercier** [comment]
> pkg.json seem to have a typo

The issue has been fixed. Can you mark your change request as resolved to unlock merging?

**Review by Evan-Adam — APPROVED** [review]

**Review by Evan-Adam — APPROVED** [review]

**Review by tuntoja — APPROVED** [review]

**Pull request merged**